### DISCUSSION TITLE
Error building base tools

### DISCUSSION THREAD

**Discussion by hyperswine** [tianocore/edk2#2889] [Q&A]
I cd into Basetools and ran `make`. This is what I got:

```
error: unknown warning option '-Werror=stringop-overflow'; did you mean '-Werror=shift-overflow'? [-Werror,-Wunknown-warning-option]
```

I am on an m1 mac with gcc 12/clang 13 and tried both. Was the warning option removed in the new compilers versions?

**Comment by osteffenrh** [comment]
That is strange. What branch are you on?

Just checked: It works with GCC 12 on Fedora 36, EDK2 master branch.

**Comment by lersi** [comment]
try to use older version of EDK2, the newest version that worked for me (Mac book air 2019 i5) is `edk2-stable202111`

also I made a little project that will help you compile edk2 application.

repo link: [edk2_project_template](https://github.com/lersi/edk2_project_template/)
